toolkit/tools/imagegen/installutils/installutils.go

```diff
@@ -781,36 +782,109 @@ func calculateTotalPackages(packages []string, installRoot string) (installedPac
+	var otherFilesToRemove = []string{
+		"/var/lib/systemd/random-seed",
+		"/boot/efi/loader/random-seed",
+		"/var/lib/systemd/credential.secret",
+	}
+
+	// From https://www.freedesktop.org/software/systemd/man/latest/machine-id.html#Initialization:
 	// For operating system images which are created once and used on multiple
 	// machines, for example for containers or in the cloud, /etc/machine-id
 	// should be either missing or an empty file in the generic file system
 	// image (the difference between the two options is described under
-	//"First Boot Semantics" below). An ID will be generated during boot and
+	// "First Boot Semantics" below). An ID will be generated during boot and
 	// saved to this file if possible. Having an empty file in place is useful
 	// because it allows a temporary file to be bind-mounted over the real file,
 	// in case the image is used read-only
-
-	const (
-		machineIDFile      = "/etc/machine-id"
-		machineIDFilePerms = 0444
-	)
-
-	ReportAction("Configuring machine id")
-
+	//
+	// From https://www.freedesktop.org/software/systemd/man/latest/machine-id.html#First%20Boot%20Semantics:
+	//     etc/machine-id is used to decide whether a boot is the first one. The rules are as follows:
+	//     1. The kernel command argument systemd.condition-first-boot= may be used to override the autodetection logic,
+	//         see kernel-command-line(7).
+	//     2. Otherwise, if /etc/machine-id does not exist, this is a first boot. During early boot, systemd will write
+	//         "uninitialized\n" to this file and overmount a temporary file which contains the actual machine ID. Later
+	//         (after first-boot-complete.target has been reached), the real machine ID will be written to disk.
+	//     3. If /etc/machine-id contains the string "uninitialized", a boot is also considered the first boot. The same
+	//         mechanism as above applies.
+	//     4. If /etc/machine-id exists and is empty, a boot is not considered the first boot. systemd will still
+	//         bind-mount a file containing the actual machine-id over it and later try to commit it to disk (if /etc/ is
+	//         writable).
+	//     5. If /etc/machine-id already contains a valid machine-id, this is not a first boot.
+	//     If according to the above rules a first boot is detected, units with ConditionFirstBoot=yes will be run and
+	//     systemd will perform additional initialization steps, in particular presetting units.
+	//
+	// However, the detailed guide for creating images (https://systemd.io/BUILDING_IMAGES/) suggests that the
+	// machine-id file should be "uninitialized\n" to trigger first-boot mode. This is because the machine-id file is
+	// used to determine if the image is being booted for the first time. If the file is empty, systemd will assume it
+	// is not the first boot and will not perform first-boot actions.
+	// (see https://www.freedesktop.org/software/systemd/man/latest/systemd-firstboot.html)
```

Comment:
```suggestion
	// We will use option 4) by default since AZL has traditionally not used firstboot mechanisms. All configuration
	// that systemd-firstboot would set should have already been configured by the imager tool. It is important to
	// create an empty file so that read-only configurations will work as expected. If the user requests that firstboot
	// be enabled we will set it to "uninitalized" as per option 3).
```